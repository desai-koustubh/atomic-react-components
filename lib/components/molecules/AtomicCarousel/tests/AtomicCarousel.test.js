import React from 'react';
import { mount } from 'enzyme';
import { AtomicCarouselVanilla } from '..';

describe('<AtomicCarousel />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(
      <AtomicCarouselVanilla>
        <h1>Test1</h1>
        <h1>Test2</h1>
        <h1>Test3</h1>
        <h1>Test4</h1>
      </AtomicCarouselVanilla>
    );
  });

  it('Should render correctly', () => {
    expect(wrapper.exists('.atomic-dots')).toBe(true);
    expect(wrapper.exists('.previous')).toBe(true);
    expect(wrapper.exists('.next')).toBe(true);
    expect(wrapper.find('.carousel-slides').contains(<h1>Test1</h1>)).toEqual(true);
    expect(wrapper).toMatchSnapshot();
  });

  it('Should render dots correctly', () => {
    wrapper.setProps({ dots: false });
    expect(wrapper.exists('.atomic-dots')).toBe(false);
  });

  it('Should render navigation correctly', () => {
    wrapper.setProps({ navigation: false });
    expect(wrapper.exists('.previous')).toBe(false);
    expect(wrapper.exists('.next')).toBe(false);
  });

  it('Should increase slideIndex on click next', () => {
    wrapper
      .find('button')
      .find('.next')
      .simulate('click');
    expect(wrapper.find('.carousel-slides').contains(<h1>Test2</h1>)).toEqual(true);
  });

  it('Should increase slideIndex on key press next', () => {
    wrapper
      .find('button')
      .find('.next')
      .simulate('keyUp', { key: 'ArrowRight' });
    expect(wrapper.find('.carousel-slides').contains(<h1>Test2</h1>)).toEqual(true);
  });

  it('Should decrease slideIndex on click prev', () => {
    wrapper
      .find('button')
      .find('.next')
      .simulate('click');
    wrapper
      .find('button')
      .find('.next')
      .simulate('click');
    wrapper
      .find('button')
      .find('.previous')
      .simulate('click');
    expect(wrapper.find('.carousel-slides').contains(<h1>Test2</h1>)).toEqual(true);
  });

  it('Should decrease slideIndex on key press prev', () => {
    wrapper
      .find('button')
      .find('.next')
      .simulate('keyUp', { key: 'ArrowRight' });
    expect(wrapper.find('.carousel-slides').contains(<h1>Test2</h1>)).toEqual(true);
  });

  it('Should show specific slideIndex on click specific dot', () => {
    wrapper
      .find('.atomic-dots')
      .find('Button')
      .at(2)
      .find('button')
      .simulate('click');
    expect(wrapper.find('.carousel-slides').contains(<h1>Test3</h1>)).toEqual(true);
  });

  it('Should show required number of slides', () => {
    const slideWrapper = jest
      .fn()
      .mockImplementation(slide => <div className="slide-item">{slide}</div>);
    wrapper.setProps({ perSlideFrame: 2 });
    expect(
      wrapper
        .find('.carousel-slides')
        .contains([slideWrapper(<h1>Test1</h1>), slideWrapper(<h1>Test2</h1>)])
    ).toEqual(true);
  });

  it('Should render next button Label correctly', () => {
    wrapper.setProps({ nextLabel: 'Next Item' });
    expect(
      wrapper
        .find('button')
        .find('.next')
        .text()
    ).toEqual('Next Item');
  });

  it('Should render previous button Label correctly', () => {
    wrapper.setProps({ prevLabel: 'Previous Item' });
    expect(
      wrapper
        .find('button')
        .find('.previous')
        .text()
    ).toEqual('Previous Item');
  });

  it('Should render heading text correctly', () => {
    wrapper.setProps({ heading: 'Featured Carousel' });
    expect(
      wrapper
        .find('.carousel-heading')
        .at(1)
        .text()
    ).toEqual('Featured Carousel');
  });

  it('Should render header correctly', () => {
    wrapper.setProps({ displayHeader: false });
    expect(wrapper.exists('#carousel')).toEqual(false);
  });

  it('Should render play/stop label correctly', () => {
    wrapper.setProps({ stopLabel: 'Stop Button' });
    wrapper.setProps({ playLabel: 'Play Button' });
    expect(
      wrapper
        .find('.animation-panel')
        .find('button')
        .text()
    ).toEqual('Stop Button');
    wrapper.unmount();
    wrapper.setProps({ autoPlay: false });
    wrapper.mount();
    expect(
      wrapper
        .find('.animation-panel')
        .find('button')
        .text()
    ).toEqual('Play Button');
  });

  it('Should render animation correctly', () => {
    const slideTransitionTime = 2000;
    wrapper.setProps({ slideTransitionTime });
    setTimeout(() => {
      expect(wrapper.find('.carousel-slides').contains(<h1>Test1</h1>)).toEqual(true);
    }, slideTransitionTime * 2);
  });

  it('Should stop rendering on mouse enter', () => {
    const slideTransitionTime = 2000;
    wrapper.setProps({ slideTransitionTime });
    wrapper.find('.carousel-slides').simulate('mouseenter');
    setTimeout(() => {
      expect(wrapper.find('.carousel-slides').contains(<h1>Test1</h1>)).toEqual(true);
      done();
    }, slideTransitionTime);
    setTimeout(() => {
      expect(wrapper.find('.carousel-slides').contains(<h1>Test1</h1>)).toEqual(true);
      done();
    }, slideTransitionTime * 2);
  });
});
