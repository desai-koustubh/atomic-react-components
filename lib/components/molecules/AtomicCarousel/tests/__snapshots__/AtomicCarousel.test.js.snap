// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AtomicCarousel /> Should render correctly 1`] = `
<AtomicCarousel
  animationIcon={true}
  autoPlay={true}
  classNames=""
  displayHeader={true}
  dots={true}
  fade={true}
  heading="Carousel"
  navigation={true}
  navigationIcons={true}
  nextLabel="next"
  perSlideFrame={1}
  playLabel="play"
  prevLabel="prev"
  slideTransitionTime={1500}
  stopLabel="stop"
>
  <section
    aria-labelledby="carousel"
    className="carousel"
  >
    <header
      id="carousel"
    >
      <Heading
        HeadingType="h1"
        className="carousel-heading"
        elementLocator=""
        inheritedStyles=""
      >
        <h1
          className="carousel-heading"
          data-locator="heading-h1-carousel"
        >
          Carousel
        </h1>
      </Heading>
    </header>
    <div
      className="carousel-container"
    >
      <Button
        ariaLabel=""
        className="previous navigation-image prev-icon"
        data-action="prev"
        disabled={true}
        elementLocator=""
        inheritedStyles=""
        onClick={[Function]}
        onKeyUp={[Function]}
        primary={false}
        secondary={false}
        tertiary={false}
        title="prev"
        type="button"
      >
        <button
          aria-label={null}
          className="previous navigation-image prev-icon"
          data-action="prev"
          data-locator="button-button-prev"
          disabled={true}
          onClick={[Function]}
          onKeyUp={[Function]}
          title="prev"
          type="button"
        >
          prev
        </button>
      </Button>
      <div
        className="carousel-slides fadeIn"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          className="slide-item"
        >
          <h1>
            Test1
          </h1>
        </div>
      </div>
      <Button
        ariaLabel=""
        className="next navigation-image next-icon"
        data-action="next"
        disabled={false}
        elementLocator=""
        inheritedStyles=""
        onClick={[Function]}
        onKeyUp={[Function]}
        primary={false}
        secondary={false}
        tertiary={false}
        title="next"
        type="button"
      >
        <button
          aria-label={null}
          className="next navigation-image next-icon"
          data-action="next"
          data-locator="button-button-next"
          disabled={false}
          onClick={[Function]}
          onKeyUp={[Function]}
          title="next"
          type="button"
        >
          next
        </button>
      </Button>
    </div>
    <div
      className="atomic-dots"
    >
      <List
        ListType="ul"
        borderSeparator={false}
        className=""
        inheritedStyles=""
        inline={false}
        nested={false}
        noDefaultView={false}
        noMargin={false}
      >
        <ul
          borderSeparator={false}
          className=""
          inline={false}
          nested={false}
          noDefaultView={false}
          noMargin={false}
          style={String {}}
        >
          <li
            className="atomic-item active"
            key="1"
          >
            <Button
              ariaLabel=""
              className=""
              disabled={false}
              elementLocator=""
              inheritedStyles=""
              onClick={[Function]}
              onKeyUp={[Function]}
              primary={false}
              secondary={false}
              tertiary={false}
              type="button"
            >
              <button
                aria-label={null}
                className=""
                data-locator="button-button-"
                disabled={false}
                onClick={[Function]}
                onKeyUp={[Function]}
                type="button"
              >
                1
              </button>
            </Button>
          </li>
          <li
            className="atomic-item"
            key="2"
          >
            <Button
              ariaLabel=""
              className=""
              disabled={false}
              elementLocator=""
              inheritedStyles=""
              onClick={[Function]}
              onKeyUp={[Function]}
              primary={false}
              secondary={false}
              tertiary={false}
              type="button"
            >
              <button
                aria-label={null}
                className=""
                data-locator="button-button-"
                disabled={false}
                onClick={[Function]}
                onKeyUp={[Function]}
                type="button"
              >
                2
              </button>
            </Button>
          </li>
          <li
            className="atomic-item"
            key="3"
          >
            <Button
              ariaLabel=""
              className=""
              disabled={false}
              elementLocator=""
              inheritedStyles=""
              onClick={[Function]}
              onKeyUp={[Function]}
              primary={false}
              secondary={false}
              tertiary={false}
              type="button"
            >
              <button
                aria-label={null}
                className=""
                data-locator="button-button-"
                disabled={false}
                onClick={[Function]}
                onKeyUp={[Function]}
                type="button"
              >
                3
              </button>
            </Button>
          </li>
          <li
            className="atomic-item"
            key="4"
          >
            <Button
              ariaLabel=""
              className=""
              disabled={false}
              elementLocator=""
              inheritedStyles=""
              onClick={[Function]}
              onKeyUp={[Function]}
              primary={false}
              secondary={false}
              tertiary={false}
              type="button"
            >
              <button
                aria-label={null}
                className=""
                data-locator="button-button-"
                disabled={false}
                onClick={[Function]}
                onKeyUp={[Function]}
                type="button"
              >
                4
              </button>
            </Button>
          </li>
        </ul>
      </List>
    </div>
    <div
      aria-label="animation"
      className="animation-panel"
    >
      <Button
        ariaLabel=""
        className="animation-btn-icon stop"
        disabled={false}
        elementLocator=""
        inheritedStyles=""
        onClick={[Function]}
        primary={false}
        secondary={false}
        tertiary={false}
        type="button"
      >
        <button
          aria-label={null}
          className="animation-btn-icon stop"
          data-locator="button-button-stop"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          stop
        </button>
      </Button>
    </div>
    <div
      aria-atomic="true"
      aria-live="polite"
      className="hidden"
      tabIndex="-1"
    >
      Item 1 of 4
    </div>
  </section>
</AtomicCarousel>
`;
